### ISSUE TITLE
set a hint in 1.7.5 that python 2 becomes deprecated

### ISSUE THREAD

**Issue by ReimarBauer** [Open-MSS/MSS#381]
**[Original report](https://bitbucket.org/wxmetvis/mss/issue/381) by Shivashis Padhi (Bitbucket: [plant99](https://bitbucket.org/plant99), GitHub: [plant99](https://github.com/plant99)).**

----------------------------------------

After following the steps [here](https://mss.readthedocs.io/en/stable/development.html) , ```test_open_rs``` at line 115, ```mss/mslib/msui/tests/test_topview.py``` fails. Below is the related error log generated. 

```

______________________ Test_MSSTopViewWindow.test_open_rs ______________________

self = <test_topview.Test_MSSTopViewWindow object at 0x7f4943522710>
mockcrit = <MagicMock name='QMessageBox' id='139951676725968'>

    @mock.patch("mslib.msui.mss_qt.QtWidgets.QMessageBox")
    def test_open_rs(self, mockcrit):
        self.window.cbTools.currentIndexChanged.emit(3)
        QtWidgets.QApplication.processEvents()
        rsdock = self.window.docks[2].widget()
        QtWidgets.QApplication.processEvents()
        QtTest.QTest.mouseClick(rsdock.cbDrawTangents, QtCore.Qt.LeftButton)
        QtWidgets.QApplication.processEvents()
        rsdock.dsbTangentHeight.setValue(6)
        QtWidgets.QApplication.processEvents()
        rsdock.dsbObsAngleAzimuth.setValue(70)
        QtTest.QTest.mouseClick(rsdock.cbDrawTangents, QtCore.Qt.LeftButton)
        QtWidgets.QApplication.processEvents()
        rsdock.cbShowSolarAngle.setChecked(True)
>       assert mockcrit.critical.call_count == 0
E       AssertionError: assert 3 == 0
E        +  where 3 = <MagicMock name='QMessageBox.critical' id='139951675972560'>.call_count
E        +    where <MagicMock name='QMessageBox.critical' id='139951675972560'> = <MagicMock name='QMessageBox' id='139951676725968'>.critical

mslib/msui/_tests/test_topview.py:115: AssertionError
----------------------------- Captured stderr call -----------------------------
[#################################] 100% de421.bsp
[#################################] 100% deltat.data
[#################################] 100% deltat.preds
[#################################] 100% Leap_Second.dat
Traceback (most recent call last):
  File "/home/darthshiv/Documents/code/mss/mslib/msui/remotesensing_dockwidget.py", line 129, in update_settings
    self.view.set_remote_sensing_appearance(settings)
  File "/home/darthshiv/Documents/code/mss/mslib/msui/mpl_qtwidget.py", line 1100, in set_remote_sensing_appearance
    self.waypoints_interactor.redraw_path()
  File "/home/darthshiv/Documents/code/mss/mslib/msui/mpl_pathinteractor.py", line 946, in redraw_path
    self.map, wp_vertices, wp_heights)
  File "/home/darthshiv/Documents/code/mss/mslib/msui/remotesensing_dockwidget.py", line 165, in compute_tangent_lines
    cut_height=self.dsbTangentHeight.value()) for i in range(len(fine_lines))]
  File "/home/darthshiv/Documents/code/mss/mslib/msui/remotesensing_dockwidget.py", line 278, in tangent_point_coordinates
    if isinstance(flight_alt, (collections.abc.Sequence, np.ndarray)):
AttributeError: 'module' object has no attribute 'abc'
CRITICAL:root:MSS Version: 1.7.4
CRITICAL:root:Python Version: 2.7.15 |Anaconda, Inc.| (default, Dec 14 2018, 19:04:19) 
[GCC 7.3.0]
CRITICAL:root:Platform: Linux-4.15.0-43-generic-x86_64-with-debian-buster-sid (('64bit', ''))
CRITICAL:root:Fatal error: Traceback (most recent call last):
  File "/home/darthshiv/Documents/code/mss/mslib/msui/remotesensing_dockwidget.py", line 129, in update_settings
    self.view.set_remote_sensing_appearance(settings)
  File "/home/darthshiv/Documents/code/mss/mslib/msui/mpl_qtwidget.py", line 1100, in set_remote_sensing_appearance
    self.waypoints_interactor.redraw_path()
  File "/home/darthshiv/Documents/code/mss/mslib/msui/mpl_pathinteractor.py", line 946, in redraw_path
    self.map, wp_vertices, wp_heights)
  File "/home/darthshiv/Documents/code/mss/mslib/msui/remotesensing_dockwidget.py", line 165, in compute_tangent_lines
    cut_height=self.dsbTangentHeight.value()) for i in range(len(fine_lines))]
  File "/home/darthshiv/Documents/code/mss/mslib/msui/remotesensing_dockwidget.py", line 278, in tangent_point_coordinates
    if isinstance(flight_alt, (collections.abc.Sequence, np.ndarray)):
AttributeError: 'module' object has no attribute 'abc'

Traceback (most recent call last):
  File "/home/darthshiv/Documents/code/mss/mslib/msui/remotesensing_dockwidget.py", line 129, in update_settings
    self.view.set_remote_sensing_appearance(settings)
  File "/home/darthshiv/Documents/code/mss/mslib/msui/mpl_qtwidget.py", line 1100, in set_remote_sensing_appearance
    self.waypoints_interactor.redraw_path()
  File "/home/darthshiv/Documents/code/mss/mslib/msui/mpl_pathinteractor.py", line 946, in redraw_path
    self.map, wp_vertices, wp_heights)
  File "/home/darthshiv/Documents/code/mss/mslib/msui/remotesensing_dockwidget.py", line 165, in compute_tangent_lines
    cut_height=self.dsbTangentHeight.value()) for i in range(len(fine_lines))]
  File "/home/darthshiv/Documents/code/mss/mslib/msui/remotesensing_dockwidget.py", line 278, in tangent_point_coordinates
    if isinstance(flight_alt, (collections.abc.Sequence, np.ndarray)):
AttributeError: 'module' object has no attribute 'abc'
CRITICAL:root:MSS Version: 1.7.4
CRITICAL:root:Python Version: 2.7.15 |Anaconda, Inc.| (default, Dec 14 2018, 19:04:19) 
[GCC 7.3.0]
CRITICAL:root:Platform: Linux-4.15.0-43-generic-x86_64-with-debian-buster-sid (('64bit', ''))
CRITICAL:root:Fatal error: Traceback (most recent call last):
  File "/home/darthshiv/Documents/code/mss/mslib/msui/remotesensing_dockwidget.py", line 129, in update_settings
    self.view.set_remote_sensing_appearance(settings)
  File "/home/darthshiv/Documents/code/mss/mslib/msui/mpl_qtwidget.py", line 1100, in set_remote_sensing_appearance
    self.waypoints_interactor.redraw_path()
  File "/home/darthshiv/Documents/code/mss/mslib/msui/mpl_pathinteractor.py", line 946, in redraw_path
    self.map, wp_vertices, wp_heights)
  File "/home/darthshiv/Documents/code/mss/mslib/msui/remotesensing_dockwidget.py", line 165, in compute_tangent_lines
    cut_height=self.dsbTangentHeight.value()) for i in range(len(fine_lines))]
  File "/home/darthshiv/Documents/code/mss/mslib/msui/remotesensing_dockwidget.py", line 278, in tangent_point_coordinates
    if isinstance(flight_alt, (collections.abc.Sequence, np.ndarray)):
AttributeError: 'module' object has no attribute 'abc'

Traceback (most recent call last):
  File "/home/darthshiv/Documents/code/mss/mslib/msui/remotesensing_dockwidget.py", line 129, in update_settings
    self.view.set_remote_sensing_appearance(settings)
  File "/home/darthshiv/Documents/code/mss/mslib/msui/mpl_qtwidget.py", line 1100, in set_remote_sensing_appearance
    self.waypoints_interactor.redraw_path()
  File "/home/darthshiv/Documents/code/mss/mslib/msui/mpl_pathinteractor.py", line 946, in redraw_path
    self.map, wp_vertices, wp_heights)
  File "/home/darthshiv/Documents/code/mss/mslib/msui/remotesensing_dockwidget.py", line 165, in compute_tangent_lines
    cut_height=self.dsbTangentHeight.value()) for i in range(len(fine_lines))]
  File "/home/darthshiv/Documents/code/mss/mslib/msui/remotesensing_dockwidget.py", line 278, in tangent_point_coordinates
    if isinstance(flight_alt, (collections.abc.Sequence, np.ndarray)):
AttributeError: 'module' object has no attribute 'abc'
CRITICAL:root:MSS Version: 1.7.4
CRITICAL:root:Python Version: 2.7.15 |Anaconda, Inc.| (default, Dec 14 2018, 19:04:19) 
[GCC 7.3.0]
CRITICAL:root:Platform: Linux-4.15.0-43-generic-x86_64-with-debian-buster-sid (('64bit', ''))
CRITICAL:root:Fatal error: Traceback (most recent call last):
  File "/home/darthshiv/Documents/code/mss/mslib/msui/remotesensing_dockwidget.py", line 129, in update_settings
    self.view.set_remote_sensing_appearance(settings)
  File "/home/darthshiv/Documents/code/mss/mslib/msui/mpl_qtwidget.py", line 1100, in set_remote_sensing_appearance
    self.waypoints_interactor.redraw_path()
  File "/home/darthshiv/Documents/code/mss/mslib/msui/mpl_pathinteractor.py", line 946, in redraw_path
    self.map, wp_vertices, wp_heights)
  File "/home/darthshiv/Documents/code/mss/mslib/msui/remotesensing_dockwidget.py", line 165, in compute_tangent_lines
    cut_height=self.dsbTangentHeight.value()) for i in range(len(fine_lines))]
  File "/home/darthshiv/Documents/code/mss/mslib/msui/remotesensing_dockwidget.py", line 278, in tangent_point_coordinates
    if isinstance(flight_alt, (collections.abc.Sequence, np.ndarray)):
AttributeError: 'module' object has no attribute 'abc'

------------------------------ Captured log call -------------------------------
mss_qt.py                  169 CRITICAL MSS Version: 1.7.4
mss_qt.py                  170 CRITICAL Python Version: 2.7.15 |Anaconda, Inc.| (default, Dec 14 2018, 19:04:19) 
[GCC 7.3.0]
mss_qt.py                  171 CRITICAL Platform: Linux-4.15.0-43-generic-x86_64-with-debian-buster-sid (('64bit', ''))
mss_qt.py                  172 CRITICAL Fatal error: Traceback (most recent call last):
  File "/home/darthshiv/Documents/code/mss/mslib/msui/remotesensing_dockwidget.py", line 129, in update_settings
    self.view.set_remote_sensing_appearance(settings)
  File "/home/darthshiv/Documents/code/mss/mslib/msui/mpl_qtwidget.py", line 1100, in set_remote_sensing_appearance
    self.waypoints_interactor.redraw_path()
  File "/home/darthshiv/Documents/code/mss/mslib/msui/mpl_pathinteractor.py", line 946, in redraw_path
    self.map, wp_vertices, wp_heights)
  File "/home/darthshiv/Documents/code/mss/mslib/msui/remotesensing_dockwidget.py", line 165, in compute_tangent_lines
    cut_height=self.dsbTangentHeight.value()) for i in range(len(fine_lines))]
  File "/home/darthshiv/Documents/code/mss/mslib/msui/remotesensing_dockwidget.py", line 278, in tangent_point_coordinates
    if isinstance(flight_alt, (collections.abc.Sequence, np.ndarray)):
AttributeError: 'module' object has no attribute 'abc'

mss_qt.py                  169 CRITICAL MSS Version: 1.7.4
mss_qt.py                  170 CRITICAL Python Version: 2.7.15 |Anaconda, Inc.| (default, Dec 14 2018, 19:04:19) 
[GCC 7.3.0]
mss_qt.py                  171 CRITICAL Platform: Linux-4.15.0-43-generic-x86_64-with-debian-buster-sid (('64bit', ''))
mss_qt.py                  172 CRITICAL Fatal error: Traceback (most recent call last):
  File "/home/darthshiv/Documents/code/mss/mslib/msui/remotesensing_dockwidget.py", line 129, in update_settings
    self.view.set_remote_sensing_appearance(settings)
  File "/home/darthshiv/Documents/code/mss/mslib/msui/mpl_qtwidget.py", line 1100, in set_remote_sensing_appearance
    self.waypoints_interactor.redraw_path()
  File "/home/darthshiv/Documents/code/mss/mslib/msui/mpl_pathinteractor.py", line 946, in redraw_path
    self.map, wp_vertices, wp_heights)
  File "/home/darthshiv/Documents/code/mss/mslib/msui/remotesensing_dockwidget.py", line 165, in compute_tangent_lines
    cut_height=self.dsbTangentHeight.value()) for i in range(len(fine_lines))]
  File "/home/darthshiv/Documents/code/mss/mslib/msui/remotesensing_dockwidget.py", line 278, in tangent_point_coordinates
    if isinstance(flight_alt, (collections.abc.Sequence, np.ndarray)):
AttributeError: 'module' object has no attribute 'abc'

mss_qt.py                  169 CRITICAL MSS Version: 1.7.4
mss_qt.py                  170 CRITICAL Python Version: 2.7.15 |Anaconda, Inc.| (default, Dec 14 2018, 19:04:19) 
[GCC 7.3.0]
mss_qt.py                  171 CRITICAL Platform: Linux-4.15.0-43-generic-x86_64-with-debian-buster-sid (('64bit', ''))
mss_qt.py                  172 CRITICAL Fatal error: Traceback (most recent call last):
  File "/home/darthshiv/Documents/code/mss/mslib/msui/remotesensing_dockwidget.py", line 129, in update_settings
    self.view.set_remote_sensing_appearance(settings)
  File "/home/darthshiv/Documents/code/mss/mslib/msui/mpl_qtwidget.py", line 1100, in set_remote_sensing_appearance
    self.waypoints_interactor.redraw_path()
  File "/home/darthshiv/Documents/code/mss/mslib/msui/mpl_pathinteractor.py", line 946, in redraw_path
    self.map, wp_vertices, wp_heights)
  File "/home/darthshiv/Documents/code/mss/mslib/msui/remotesensing_dockwidget.py", line 165, in compute_tangent_lines
    cut_height=self.dsbTangentHeight.value()) for i in range(len(fine_lines))]
  File "/home/darthshiv/Documents/code/mss/mslib/msui/remotesensing_dockwidget.py", line 278, in tangent_point_coordinates
    if isinstance(flight_alt, (collections.abc.Sequence, np.ndarray)):
AttributeError: 'module' object has no attribute 'abc'

```

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

HI @plant99 
not so good start. 

Can you show 


```
#!text

conda list 
```

And also I was not aware of a py2.7 issue yet. We think about skipping py2 with the next 1.8.0 release. Can you try with anaconda3/miniconda3 ?

**Comment by ReimarBauer** [comment]
**Original comment by Shivashis Padhi (Bitbucket: [plant99](https://bitbucket.org/plant99), GitHub: [plant99](https://github.com/plant99)).**

----------------------------------------

```

# packages in environment at /media/darthshiv/a2468a4a-a1e3-4ffc-b9ac-5a20d80703e8/frodo/anaconda/envs/mssdev:
#
# Name                    Version                   Build  Channel
alabaster                 0.7.12                     py_0    conda-forge
apipkg                    1.5                        py_0    conda-forge
appdirs                   1.4.3                      py_1    conda-forge
argcomplete               1.9.4                 py27_1000    conda-forge
astropy                   2.0.9                    py27_0    conda-forge
atomicwrites              1.3.0                      py_0    conda-forge
attrs                     18.2.0                     py_0    conda-forge
babel                     2.6.0                      py_1    conda-forge
backports                 1.0                        py_2    conda-forge
backports.functools_lru_cache 1.5                        py_1    conda-forge
backports.os              0.1.1                 py27_1000    conda-forge
backports_abc             0.5                        py_1    conda-forge
basemap                   1.2.0           py27h673bf1a_1001    conda-forge
beautifulsoup4            4.7.1                 py27_1001    conda-forge
blas                      1.0                         mkl  
bzip2                     1.0.6             h14c3975_1002    conda-forge
ca-certificates           2018.11.29           ha4d7672_0    conda-forge
certifi                   2018.11.29            py27_1000    conda-forge
cftime                    1.0.3.4         py27h3010b51_1000    conda-forge
chameleon                 3.5                      py27_0  
configparser              3.7.3                    py27_1    conda-forge
coverage                  4.5.2           py27h14c3975_1001    conda-forge
curl                      7.62.0               hbc83047_0  
cycler                    0.10.0                     py_1    conda-forge
dbus                      1.13.2               h714fa37_1  
docutils                  0.14                  py27_1001    conda-forge
easyprocess               0.2.3                 py27_1000    conda-forge
ecdsa                     0.13                       py_0    conda-forge
entrypoints               0.3                   py27_1000    conda-forge
enum34                    1.1.6                 py27_1001    conda-forge
execnet                   1.5.0                      py_0    conda-forge
expat                     2.2.5             hf484d3e_1002    conda-forge
flake8                    3.7.6                    py27_0    conda-forge
fontconfig                2.13.1            h2176d3f_1000    conda-forge
freetype                  2.9.1             h94bbf69_1005    conda-forge
fs                        2.4.3                    py27_0    conda-forge
fs.sshfs                  0.6.1                    py27_0    conda-forge
fs.webdavfs               0.3.5.3                  py27_0    conda-forge
fs_filepicker             0.3.7                    py27_2    conda-forge
funcsigs                  1.0.2                      py_3    conda-forge
functools32               3.2.3.2                    py_3    conda-forge
furl                      1.0.1                    py27_0    conda-forge
future                    0.17.1                py27_1000    conda-forge
futures                   3.2.0                 py27_1000    conda-forge
geos                      3.7.1             hf484d3e_1000    conda-forge
gettext                   0.19.8.1          h9745a5d_1001    conda-forge
gitdb2                    2.0.5                      py_0    conda-forge
gitpython                 2.1.11                     py_0    conda-forge
glib                      2.56.2            had28632_1001    conda-forge
gst-plugins-base          1.14.0               hbbd80ab_1  
gstreamer                 1.14.0               hb453b48_1  
hdf4                      4.2.13            h9a582f1_1002    conda-forge
hdf5                      1.10.4          nompi_h11e915b_1105    conda-forge
html5lib                  1.0.1                      py_0    conda-forge
humanfriendly             4.18                     py27_0    conda-forge
icu                       58.2              hf484d3e_1000    conda-forge
idna                      2.8                   py27_1000    conda-forge
imagesize                 1.1.0                      py_0    conda-forge
intel-openmp              2019.1                      144  
isodate                   0.6.0                      py_1    conda-forge
jinja2                    2.10                       py_1    conda-forge
jpeg                      9c                h14c3975_1001    conda-forge
jplephem                  2.9                pyh11f86e8_0    conda-forge
kiwisolver                1.0.1           py27h6bb024c_1002    conda-forge
libcurl                   7.62.0               h20c2e04_0  
libffi                    3.2.1             hf484d3e_1005    conda-forge
libgcc-ng                 7.3.0                hdf63c60_0    conda-forge
libgfortran-ng            7.2.0                hdf63c60_3    conda-forge
libiconv                  1.15              h14c3975_1004    conda-forge
libnetcdf                 4.6.2             hbdf4f91_1001    conda-forge
libpng                    1.6.36            h84994c4_1000    conda-forge
libssh2                   1.8.0                         1    conda-forge
libstdcxx-ng              7.3.0                hdf63c60_0    conda-forge
libtiff                   4.0.10            h648cc4a_1001    conda-forge
libuuid                   2.32.1            h14c3975_1000    conda-forge
libxcb                    1.13              h14c3975_1002    conda-forge
libxml2                   2.9.8             h143f9aa_1005    conda-forge
libxslt                   1.1.32            h4785a14_1002    conda-forge
lxml                      4.3.1            py27h23eabaa_0    conda-forge
markupsafe                1.1.1            py27h14c3975_0    conda-forge
matplotlib                2.2.3            py27h8a2030e_1    conda-forge
matplotlib-base           2.2.3            py27h60b886d_1    conda-forge
mccabe                    0.6.1                      py_1    conda-forge
mkl                       2019.1                      144  
mkl_fft                   1.0.10           py27h14c3975_1    conda-forge
mkl_random                1.0.2            py27h637b7d7_2    conda-forge
mock                      2.0.0                 py27_1001    conda-forge
more-itertools            4.3.0                 py27_1000    conda-forge
mss                       1.7.4                 py27_1001    conda-forge
ncurses                   6.1               hf484d3e_1002    conda-forge
netcdf4                   1.4.2           py27had69b76_1001    conda-forge
numpy                     1.15.4           py27h7e9f1db_0  
numpy-base                1.15.4           py27hde5b4d6_0  
olefile                   0.46                       py_0    conda-forge
openssl                   1.1.1a            h14c3975_1000    conda-forge
orderedmultidict          0.7.11                   py27_0    conda-forge
owslib                    0.17.1                     py_0    conda-forge
packaging                 19.0                       py_0    conda-forge
paramiko                  1.18.2                   py27_0    conda-forge
paste                     3.0.6                 py27_1000    conda-forge
pathlib2                  2.3.3                 py27_1000    conda-forge
pbr                       5.1.2                      py_0    conda-forge
pcre                      8.42                 h439df22_0  
pep8                      1.7.1                      py_0    conda-forge
pillow                    5.4.1           py27h00a061d_1000    conda-forge
pip                       19.0.3                   py27_0    conda-forge
pluggy                    0.9.0                      py_0    conda-forge
proj4                     5.2.0             h14c3975_1001    conda-forge
pthread-stubs             0.4               h14c3975_1001    conda-forge
py                        1.8.0                      py_0    conda-forge
pycodestyle               2.5.0                      py_0    conda-forge
pycrypto                  2.6.1           py27h14c3975_1002    conda-forge
pyflakes                  2.1.0                      py_0    conda-forge
pygments                  2.3.1                      py_0    conda-forge
pyparsing                 2.3.1                      py_0    conda-forge
pyproj                    1.9.6           py27hc0953d3_1000    conda-forge
pyqt                      5.6.0           py27h13b7fb3_1008    conda-forge
pyshp                     2.1.0                      py_0    conda-forge
pytest                    4.3.0                    py27_0    conda-forge
pytest-cache              1.0                      py27_0    conda-forge
pytest-cov                2.6.1                      py_0    conda-forge
pytest-flake8             1.0.4                    py27_0    conda-forge
pytest-pep8               1.0.6                      py_1    conda-forge
python                    2.7.15               h9bab390_6  
python-dateutil           2.8.0                      py_0    conda-forge
pytz                      2018.9                     py_0    conda-forge
pyvirtualdisplay          0.2.1                 py27_1000    conda-forge
qt                        5.6.3                h8bf5577_3  
readline                  7.0               hf8c457e_1001    conda-forge
requests                  2.13.0                   py27_0    conda-forge
scandir                   1.9.0           py27h14c3975_1000    conda-forge
scipy                     1.2.1            py27h7c811a0_0  
setuptools                40.8.0                   py27_0    conda-forge
sgp4                      1.4                pyh24bf2e0_3    conda-forge
singledispatch            3.4.0.3               py27_1000    conda-forge
sip                       4.18.1          py27hf484d3e_1000    conda-forge
six                       1.12.0                py27_1000    conda-forge
skyfield                  1.10                       py_0    conda-forge
smmap2                    2.0.5                      py_0    conda-forge
snowballstemmer           1.2.1                      py_1    conda-forge
soupsieve                 1.8                      py27_0    conda-forge
sphinx                    1.8.4                    py27_0    conda-forge
sphinx_rtd_theme          0.4.3                      py_0    conda-forge
sphinxcontrib-websupport  1.1.0                      py_1    conda-forge
sqlite                    3.26.0            h67949de_1000    conda-forge
subprocess32              3.5.3            py27h14c3975_0    conda-forge
tk                        8.6.9             h84994c4_1000    conda-forge
tornado                   5.1.1           py27h14c3975_1000    conda-forge
typing                    3.6.4                    py27_2    conda-forge
unicodecsv                0.14.1                     py_1    conda-forge
webdavclient2             0.1.1                      py_0    conda-forge
webencodings              0.5.1                      py_1    conda-forge
wheel                     0.33.1                   py27_0    conda-forge
xorg-libxau               1.0.9                h14c3975_0    conda-forge
xorg-libxdmcp             1.1.2             h14c3975_1007    conda-forge
xz                        5.2.4             h14c3975_1001    conda-forge
zlib                      1.2.11            h14c3975_1004    conda-forge
```

Here is output of `conda list`.
Sure, let me try a couple more times along with anaconda-3.

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

## conda list


Most of us add the default channel on top. 


```
#!text

more .condarc 
channels:
  - defaults
  - conda-forge
```

so the list shows only those added by conda-forge which are not in the defaults

```
#!text

# packages in environment at /home/icg105/miniconda2/envs/mssenv:
#
# Name                    Version                   Build  Channel
alabaster                 0.7.12                   py27_0  
apipkg                    1.5                      py27_0  
appdirs                   1.4.3            py27h28b3542_0  
argcomplete               1.9.4                    py27_0  
asn1crypto                0.24.0                   py27_0  
astropy                   2.0.9            py27hdd07704_0  
atomicwrites              1.3.0                      py_0  
attrs                     18.2.0           py27h28b3542_0  
babel                     2.6.0                    py27_0  
backports                 1.0                      py27_1  
backports.functools_lru_cache 1.5                      py27_1  
backports.os              0.1.1                    py27_0  
backports_abc             0.5              py27h7b3c97b_0  
basemap                   1.2.0            py27h705c2d8_0  
bcrypt                    3.1.6            py27h7b6447c_0  
beautifulsoup4            4.7.1                    py27_1  
blas                      1.0                         mkl  
bzip2                     1.0.6                h14c3975_5  
ca-certificates           2019.1.23                     0  
cached-property           1.5.1                    py27_0  
certifi                   2018.11.29               py27_0  
cffi                      1.11.5           py27he75722e_1  
cftime                    1.0.3.4          py27hdd07704_0  
chameleon                 3.5                      py27_0  
chardet                   3.0.4                    py27_1  
configparser              3.7.1                    py27_0  
coverage                  4.5.2            py27h7b6447c_0  
cryptography              2.5              py27h1ba5d50_0  
curl                      7.63.0            hbc83047_1000  
cycler                    0.10.0                   py27_0  
dbus                      1.13.6               h746ee38_0  
docutils                  0.14                     py27_0  
easyprocess               0.2.3                 py27_1000    conda-forge
entrypoints               0.3                      py27_0  
enum34                    1.1.6                    py27_1  
execnet                   1.5.0                    py27_0  
expat                     2.2.6                he6710b0_0  
flake8                    3.7.6                    py27_0    conda-forge
fontconfig                2.13.0               h9420a91_0  
freetype                  2.9.1                h8a8886c_1  
fs                        2.3.0                    py27_0    conda-forge
fs.sshfs                  0.10.1                   py27_0    conda-forge
fs.webdavfs               0.3.5.3                  py27_0    conda-forge
fs_filepicker             0.3.7                    py27_2    conda-forge
funcsigs                  1.0.2            py27h83f16ab_0  
functools32               3.2.3.2                  py27_1  
furl                      2.0.0                    py27_0  
future                    0.17.1                   py27_0  
futures                   3.2.0                    py27_0  
geos                      3.6.2                heeff764_2  
gitdb2                    2.0.5                    py27_0  
gitpython                 2.1.11                   py27_0  
glib                      2.56.2               hd408876_0  
gst-plugins-base          1.14.0               hbbd80ab_1  
gstreamer                 1.14.0               hb453b48_1  
hdf4                      4.2.13               h3ca952b_2  
hdf5                      1.10.4               hb1b8bf9_0  
html5lib                  1.0.1                    py27_0  
humanfriendly             4.17                     py27_0  
icu                       58.2                 h9c2bf20_1  
idna                      2.8                      py27_0  
imagesize                 1.1.0                    py27_0  
intel-openmp              2019.1                      144  
ipaddress                 1.0.22                   py27_0  
jinja2                    2.10                     py27_0  
jpeg                      9b                   h024ee3a_2  
jplephem                  2.9                pyh11f86e8_0    conda-forge
kiwisolver                1.0.1            py27hf484d3e_0  
krb5                      1.16.1               h173b8e3_7  
libcurl                   7.63.0            h20c2e04_1000  
libedit                   3.1.20181209         hc058e9b_0  
libffi                    3.2.1                hd88cf55_4  
libgcc-ng                 8.2.0                hdf63c60_1  
libgfortran-ng            7.3.0                hdf63c60_0  
libnetcdf                 4.6.1                h11d0813_2  
libpng                    1.6.36               hbc83047_0  
libsodium                 1.0.16               h1bed415_0  
libssh2                   1.8.0                h1ba5d50_4  
libstdcxx-ng              8.2.0                hdf63c60_1  
libtiff                   4.0.10               h2733197_2  
libuuid                   1.0.3                h1bed415_2  
libxcb                    1.13                 h1bed415_1  
libxml2                   2.9.9                he19cac6_0  
libxslt                   1.1.33               h7d1a2b0_0  
lxml                      4.3.1            py27hefd8a0e_0  
markupsafe                1.1.0            py27h7b6447c_0  
matplotlib                2.2.3            py27hb69df0a_0  
mccabe                    0.6.1                    py27_1  
mkl                       2019.1                      144  
mkl_fft                   1.0.10           py27ha843d7b_0  
mkl_random                1.0.2            py27hd81dba3_0  
mock                      2.0.0                    py27_0  
more-itertools            5.0.0                    py27_0  
mss                       1.7.4                 py27_1001    conda-forge
ncurses                   6.1                  he6710b0_1  
netcdf4                   1.4.2            py27h808af73_0  
numpy                     1.15.4           py27h7e9f1db_0  
numpy-base                1.15.4           py27hde5b4d6_0  
olefile                   0.46                     py27_0  
openssl                   1.1.1a               h7b6447c_0  
orderedmultidict          1.0                      py27_0  
owslib                    0.17.0                   py27_0  
packaging                 19.0                     py27_0  
paramiko                  2.4.2                    py27_0  
paste                     3.0.6                    py27_0  
pathlib2                  2.3.3                    py27_0  
pbr                       5.1.1                    py27_0  
pcre                      8.42                 h439df22_0  
pep8                      1.7.1                    py27_0  
pillow                    5.4.1            py27h34e0f95_0  
pip                       19.0.1                   py27_0  
pluggy                    0.8.1                    py27_0  
proj4                     5.2.0                he6710b0_1  
py                        1.7.0                    py27_0  
pyasn1                    0.4.5                      py_0  
pycodestyle               2.5.0                    py27_0  
pycparser                 2.19                     py27_0  
pyflakes                  2.1.0                    py27_0  
pygments                  2.3.1                    py27_0  
pynacl                    1.3.0            py27h7b6447c_0  
pyopenssl                 19.0.0                   py27_0  
pyparsing                 2.3.1                    py27_0  
pyproj                    1.9.6            py27h14380d9_0  
pyqt                      5.9.2            py27h05f1152_2  
pyshp                     2.0.1                    py27_0  
pysocks                   1.6.8                    py27_0  
pytest                    4.2.1                    py27_0  
pytest-cache              1.0                      py27_1  
pytest-cov                2.6.1                    py27_0  
pytest-flake8             1.0.4                    py27_0    conda-forge
pytest-pep8               1.0.6                    py27_1  
python                    2.7.15               h9bab390_6  
python-dateutil           2.7.5                    py27_0  
pytz                      2018.9                   py27_0  
pyvirtualdisplay          0.2.1                 py27_1000    conda-forge
qt                        5.9.7                h5867ecd_1  
readline                  7.0                  h7b6447c_5  
requests                  2.21.0                   py27_0  
scandir                   1.9.0            py27h14c3975_0  
scipy                     1.2.1            py27h7c811a0_0  
setuptools                40.8.0                   py27_0  
sgp4                      1.4                pyh24bf2e0_3    conda-forge
singledispatch            3.4.0.3          py27h9bcb476_0  
sip                       4.19.8           py27hf484d3e_0  
six                       1.12.0                   py27_0  
skyfield                  1.10                       py_0    conda-forge
smmap2                    2.0.5                    py27_0  
snowballstemmer           1.2.1            py27h44e2768_0  
soupsieve                 1.7.1                    py27_0  
sphinx                    1.8.4                    py27_0  
sphinx_rtd_theme          0.4.2                    py27_0  
sphinxcontrib             1.0                      py27_1  
sphinxcontrib-websupport  1.1.0                    py27_1  
sqlite                    3.26.0               h7b6447c_0  
subprocess32              3.5.3            py27h7b6447c_0  
tk                        8.6.8                hbc83047_0  
tornado                   5.1.1            py27h7b6447c_0  
typing                    3.6.4                    py27_0  
unicodecsv                0.14.1           py27h5062da9_0  
urllib3                   1.24.1                   py27_0  
webdavclient2             0.1.1                      py_0    conda-forge
webencodings              0.5.1                    py27_1  
wheel                     0.32.3                   py27_0  
xz                        5.2.4                h14c3975_4  
zlib                      1.2.11               h7b6447c_3  
zstd                      1.3.7                h0b5b093_0 
```

## branches

we have two branches 

* stable: is for bug fixes only and should work with py2 and py3 currently. 
* develop: is for the next release - which than becomes the next stable.  We likly will switch to py3 only, because this is anyway needed until next year.

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

I have a flake8 failure with the *stable* branch
```
#!text

____________________ FLAKE8-check(ignoring E402 W504 F401) _____________________
/home/user/PycharmProjects/mss/mslib/msui/mss_pyui.py:332:21: E117 over-indented
```

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

We have to set a deprecation warning with 1.7.5 for Python 2

**Comment by ReimarBauer** [comment]
**Original comment by Shivashis Padhi (Bitbucket: [plant99](https://bitbucket.org/plant99), GitHub: [plant99](https://github.com/plant99)).**

----------------------------------------

Thank you @ReimarBauer  for your useful comments. I can now understand the development process of mss more clearly. I will explore more about mss while on `stable` branch, can I PR on the small indentation fix? Is it too minor to be made a PR?

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

please do so. :)

PEP8 fixes are also wanted in  a small minor PR :)

**Comment by ReimarBauer** [comment]
**Original comment by Joern Ungermann (Bitbucket: [joernu76](https://bitbucket.org/joernu76), GitHub: [joernu76](https://github.com/joernu76)).**

----------------------------------------

documentation and major modules show a deprecation warning for python 2,
fixes #381
